src/v/cluster/cluster_discovery.cc

```diff
@@ -182,18 +196,259 @@ ss::future<bool> cluster_discovery::dispatch_node_uuid_registration_to_seeds(
+    vassert(
+      _cluster_bootstrap_info_fetched,
+      "Cluster bootstrap info not fetched yet");
+
+    // Validate the returned replies and populate the brokers vector
+    bool failed = false;
+    brokers seed_brokers;
+    seed_brokers.reserve(_cluster_bootstrap_info.size());
+    for (const auto& peer : _cluster_bootstrap_info) {
+        if (
+          peer.second.version
+          != features::feature_table::get_latest_logical_version()) {
+            vlog(
+              clusterlog.error,
+              "Cluster setup error: logical version mismatch, local: {}, {}: "
+              "{}",
+              features::feature_table::get_latest_logical_version(),
+              peer.first,
+              peer.second.version);
+            failed = true;
+        }
+        if (peer.second.cluster_uuid.has_value()) {
+            vlog(
+              clusterlog.error,
+              "Cluster setup error: peer is already a member of cluster {}",
+              *peer.second.cluster_uuid);
+            failed = true;
+        }
+        check_configuration_match(failed, peer.first, peer.second);
+        seed_brokers.push_back(peer.second.broker);
+    }
+    if (failed) {
+        throw std::runtime_error(fmt_with_ctx(
+          fmt::format,
+          "Cannot bootstrap a cluster due to seed servers mismatch, check "
+          "the log above for details"));
+    }
+
+    // The other seeds will likely mostly all have -1 as their node_id if their
+    // node_id was not set explicitly. Have the returned seed broker node_id
+    // populated with indices, relying on the fact that all seed broker's have
+    // identical configuration. Note that the order is preserved from
+    // seed_servers to seed_brokers, but the latter is missing one item for
+    // self.
+    const std::vector<config::seed_server>& self_seed_servers
+      = config::node().seed_servers();
+    for (auto [ib, is]
+         = std::tuple{seed_brokers.begin(), self_seed_servers.cbegin()};
+         ib != seed_brokers.cend();
+         ++ib) {
+        while (is != self_seed_servers.cend()
+               && is->addr != ib->rpc_address()) {
+            ++is;
+        }
+        if (ib->id() == model::unassigned_node_id) {
+            vassert(
+              is != self_seed_servers.cend(),
+              "seed_brokers and seed_servers mismatch");
+            const auto idx = boost::numeric_cast<model::node_id::type>(
+              std::distance(self_seed_servers.cbegin(), is));
+            ib->replace_unassigned_node_id(model::node_id{idx});
+        }
+    }
+
+    vlog(clusterlog.debug, "Seed brokers discovered: {}", seed_brokers);
+    return seed_brokers;
+}
+
+ss::future<std::optional<node_id>>
 cluster_discovery::get_cluster_founder_node_id() {
     if (config::node().empty_seed_starts_cluster()) {
         if (config::node().seed_servers().empty()) {
-            return node_id{0};
+            co_return node_id{0};
```

Comment:
do we validate that in this case node_id configured in `redpanda.yml` matches the  one returned from here ?

Reply:
nope, I agree that it should be added